### PR TITLE
chapter 4 implement

### PR DESCRIPTION
**Pull request by Dondo722**

### CODE REVIEW

File: src/main/java/com/es/phoneshop/web/CartPageServlet.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CartPageServlet  extends HttpServlet {
```
Comment (SergeiStarikov) [inline src/main/java/com/es/phoneshop/web/CartPageServlet.java:22]: Here are (maybe) in other places - please reformat the code, remove duplicate spaces, add missing spaces: Ctrl + Alt + L in IntelliJ IDEA.

File: src/main/java/com/es/phoneshop/model/cart/Cart.java [hunk 2]
```diff
@@ -1,14 +1,15 @@
 package com.es.phoneshop.model.cart;
```
Comment (SergeiStarikov) [inline src/main/java/com/es/phoneshop/model/cart/Cart.java:1]: I don't see a link to the cart page on UI. This is a bit of a non-intuitive UX.

Comment (SergeiStarikov) [inline src/main/java/com/es/phoneshop/model/cart/Cart.java:1]: Please add unit tests for new functionality.

**Review by SergeiStarikov — APPROVED** [review]

File: src/main/java/com/es/phoneshop/model/cart/Cart.java (hunk 2 above)
Comment (SergeiStarikov) [reply src/main/java/com/es/phoneshop/model/cart/Cart.java:1]: I don't see a way to go to the cart page directly from the product list page. How can the user get to the cart page besides entering http://localhost:8080/phoneshop-servlet-api/cart in the browser address bar directly?

Comment (Dondo722) [reply src/main/java/com/es/phoneshop/model/cart/Cart.java:1]: Ohhhh, i get it .  i'll fix it in next chapter

**Pull request merged**